mm2src/coins/eth/eth_rpc.rs

```diff
@@ -35,11 +35,18 @@ impl EthCoin {
                     clients.rotate_left(i);
                     return Ok(r);
                 },
-                Ok(Err(rpc_error)) => {
-                    debug!("Request on '{method}' failed. Error: {rpc_error}");
-
-                    if let Web3Transport::Websocket(socket_transport) = client.web3.transport() {
-                        socket_transport.stop_connection_loop().await;
+                Ok(Err(error)) => {
+                    debug!("Request on '{method}' failed. Error: {error}");
+
+                    match error {
+                        web3::Error::Rpc(rpc_error) => {
```

Comment:
Does this revert this fix https://github.com/KomodoPlatform/komodo-defi-framework/pull/2058#discussion_r1491715014 ? Maybe we should try next node on `ErrorCode::ServerError`.

Reply:
8f45563cfaeabd0567abdafdb93b96d8725d51cc should cover both cases.